airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/DbtLauncherWorker.java

```diff
@@ -0,0 +1,137 @@
+
+import io.airbyte.commons.io.LineGobbler;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.logging.LoggingHelper;
+import io.airbyte.commons.logging.MdcScope;
+import io.airbyte.config.OperatorDbtInput;
+import io.airbyte.scheduler.models.IntegrationLauncherConfig;
+import io.airbyte.scheduler.models.JobRunConfig;
+import io.airbyte.workers.Worker;
+import io.airbyte.workers.WorkerApp;
+import io.airbyte.workers.WorkerConfigs;
+import io.airbyte.workers.WorkerException;
+import io.airbyte.workers.WorkerUtils;
+import io.airbyte.workers.process.KubeProcessFactory;
+import io.airbyte.workers.process.ProcessFactory;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// todo: DRY the launchers
+public class DbtLauncherWorker implements Worker<OperatorDbtInput, Void> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DbtLauncherWorker.class);
+
+  private static final MdcScope.Builder LOG_MDC_BUILDER = new MdcScope.Builder()
+      .setLogPrefix("dbt-orchestrator")
+      .setPrefixColor(LoggingHelper.Color.CYAN_BACKGROUND);
+
+  public static final String DBT = "dbt";
+  public static final String INIT_FILE_DESTINATION_LAUNCHER_CONFIG = "destinationLauncherConfig.json";
+
+  private final WorkerConfigs workerConfigs;
+  private final ProcessFactory processFactory;
+  private final String airbyteVersion;
+  private final Path workspaceRoot;
+  private final IntegrationLauncherConfig destinationLauncherConfig;
+  private final JobRunConfig jobRunConfig;
+
+  private final AtomicBoolean cancelled = new AtomicBoolean(false);
+
+  private Process process;
+
+  public DbtLauncherWorker(
+                           final Path workspaceRoot,
+                           final IntegrationLauncherConfig destinationLauncherConfig,
+                           final JobRunConfig jobRunConfig,
+                           final WorkerConfigs workerConfigs,
+                           final ProcessFactory processFactory,
+                           final String airbyteVersion) {
+    this.workspaceRoot = workspaceRoot;
+    this.destinationLauncherConfig = destinationLauncherConfig;
+    this.jobRunConfig = jobRunConfig;
+    this.workerConfigs = workerConfigs;
+    this.processFactory = processFactory;
+    this.airbyteVersion = airbyteVersion;
+  }
+
+  @Override
+  public Void run(OperatorDbtInput operatorDbtInput, Path jobRoot) throws WorkerException {
+    try {
+      final Path jobPath = WorkerUtils.getJobRoot(workspaceRoot, jobRunConfig.getJobId(), jobRunConfig.getAttemptId());
+
+      // we want to filter down to remove secrets, so we aren't writing over a bunch of unnecessary
+      // secrets
+      final Map<String, String> envMap = System.getenv().entrySet().stream()
```

Comment:
okay if we DRY in a follow up PR. The important one looks to be the secrets stuff. Have to make sure we don't forget it.